client/core/core.go

```diff
@@ -6557,3 +6559,203 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
 
 	return wallet.LogFilePath()
 }
+
+func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFile *os.File) error {
+	c.wg.Add(1)
+	c.wg.Done()
```

Comment:
wg copy pasta?

Reply:
should be defer

Reply:
I don't see the goroutine though

Reply:
It's possible that the caller run this in a goroutine?

Reply:
I can just comment and take out.

Looking at the current rpcserver code, I think it depends on if go-chi will wait for the handler to finish on its server shutdown method.

Reply:
Removed.